pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentDictionaryCreator.java

```diff
@@ -154,24 +155,18 @@ public void build()
         _stringValueToIndexMap = new Object2IntOpenHashMap<>(numValues);
 
         // Get the maximum length of all entries
-        byte[][] sortedStringBytes = new byte[numValues][];
         for (int i = 0; i < numValues; i++) {
           String value = sortedStrings[i];
           _stringValueToIndexMap.put(value, i);
-          byte[] valueBytes = StringUtil.encodeUtf8(value);
-          sortedStringBytes[i] = valueBytes;
-          _numBytesPerEntry = Math.max(_numBytesPerEntry, valueBytes.length);
         }
 
         // Backward-compatible: index file is always big-endian
-        try (PinotDataBuffer dataBuffer = PinotDataBuffer
-            .mapFile(_dictionaryFile, false, 0, (long) numValues * _numBytesPerEntry, ByteOrder.BIG_ENDIAN,
+        long size = VarLengthStringsReaderWriter.getRequiredSize(sortedStrings);
```

Comment:
we need to support both modes.  We need the new implementation for byte and string column and only when the standard deviation is high.

Reply:
Are you saying we calculate the stdDev and decide to use whether fixed or var length dict based on the std dev on-the-fly?